zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/operation/AbstractOperationHandler.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.handlers.operation;
+
+import io.camunda.exporter.exceptions.PersistenceException;
+import io.camunda.exporter.handlers.ExportHandler;
+import io.camunda.exporter.store.BatchRequest;
+import io.camunda.webapps.schema.descriptors.operate.template.OperationTemplate;
+import io.camunda.webapps.schema.entities.operation.OperationEntity;
+import io.camunda.webapps.schema.entities.operation.OperationState;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordValue;
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public abstract class AbstractOperationHandler<R extends RecordValue>
+    implements ExportHandler<OperationEntity, R> {
+
+  protected final String indexName;
+
+  public AbstractOperationHandler(final String indexName) {
+    this.indexName = indexName;
+  }
+
+  @Override
+  public Class<OperationEntity> getEntityType() {
+    return OperationEntity.class;
+  }
+
+  @Override
+  public List<String> generateIds(final Record<R> record) {
+    final long operationReference = record.getOperationReference();
+    if (operationReference > 0) {
+      return List.of(String.valueOf(operationReference));
+    }
+    return List.of();
+  }
+
+  @Override
+  public OperationEntity createNewEntity(final String id) {
+    return new OperationEntity().setId(id);
+  }
+
+  @Override
+  public void updateEntity(final Record<R> record, final OperationEntity entity) {
```

Comment:
❓ Will this only be called when generateIds returns an id?

Reply:
yes, in ExporterBatchWriter, we call generateIds() and then we loop on the result to update them. if no id was generated, the update won't happen.

https://github.com/camunda/camunda/blob/main/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/ExporterBatchWriter.java#L36-L37